crates/fuel-core/src/coins_query.rs

```diff
@@ -221,6 +274,168 @@ pub async fn random_improve(
     Ok(coins_per_asset)
 }
 
+pub async fn select_coins_to_spend(
+    CoinsToSpendIndexIter {
+        big_coins_iter,
+        dust_coins_iter,
+    }: CoinsToSpendIndexIter<'_>,
+    total: u64,
+    max: u16,
+    excluded_ids: &ExcludedCoinIds<'_>,
+    batch_size: usize,
+) -> Result<Vec<CoinsToSpendIndexEntry>, CoinsQueryError> {
+    const TOTAL_AMOUNT_ADJUSTMENT_FACTOR: u64 = 2;
+    if total == 0 && max == 0 {
```

Comment:
Based on the current logic, it will return a strange error to the user. Maybe we want to return more user-friendly error here=) Or an empty result, but `coins_per_asset.is_empty()` will not allow it in the current implementation

Reply:
A proper error has been added in 2c4006939925e227771ee2a2bd800bbf3bc1be5c

Reply:
Also, a little fix to the `if` together with improved error messages has been added in bef1548de9e7dffbdeac4cd0b19eb420baa6a56f